packages/client/test/integration/SubscriberKeyExchange.test.ts

```diff
@@ -107,9 +113,52 @@ describe('SubscriberKeyExchange', () => {
             const request = await environment.getNetwork().waitForSentMessage({
                 messageType: StreamMessageType.GROUP_KEY_REQUEST
             })
-            await assertGroupKeyRequest(request, [groupKey.id])
+            await assertGroupKeyRequest(
+                request,
+                [groupKey.id],
+                toEthereumAddress(subscriberWallet.address),
+                SignatureType.SECP256K1
+            )
             const keyStore = getLocalGroupKeyStore(toEthereumAddress(subscriberWallet.address))
             await waitForCondition(async () => (await keyStore.get(groupKey.id, toEthereumAddress(publisherWallet.address))) !== undefined)
         })
+
+        it('happy path: ERC-1271', async () => {
+            const subscriber2Wallet = fastWallet()
+            const erc1271Contract = randomEthereumAddress()
+            await subscriber.grantPermissions(StreamPartIDUtils.getStreamID(streamPartId), {
+                permissions: [StreamPermission.SUBSCRIBE],
+                user: erc1271Contract
+            })
+            environment.getChain().erc1271AllowedAddresses.add(erc1271Contract, toEthereumAddress(subscriber2Wallet.address))
+
+            const groupKey = GroupKey.generate()
+            const publisher = environment.createClient({
+                auth: {
+                    privateKey: publisherWallet.privateKey
+                }
+            })
+            await publisher.addEncryptionKey(groupKey, toEthereumAddress(publisherWallet.address))
+
+            const subscriber2 = environment.createClient({
+                auth: {
+                    privateKey: subscriber2Wallet.privateKey
+                }
+            })
+            await subscriber2.subscribe({
+                id: StreamPartIDUtils.getStreamID(streamPartId),
+                partition: StreamPartIDUtils.getStreamPartition(streamPartId),
+                erc1271Contract
+            }, () => {})
+
+            await triggerGroupKeyRequest(groupKey, publisher)
+
+            const request = await environment.getNetwork().waitForSentMessage({
+                messageType: StreamMessageType.GROUP_KEY_REQUEST
+            })
+            await assertGroupKeyRequest(request, [groupKey.id], erc1271Contract, SignatureType.ERC_1271)
+            const keyStore = getLocalGroupKeyStore(await subscriber2.getAddress())
+            await waitForCondition(async () => (await keyStore.get(groupKey.id, toEthereumAddress(publisherWallet.address))) !== undefined)
```

Comment:
Could use the `subscriber` variable which we initialize in `beforeEach` instead of creating explicit `subscriber2` variable. The value of that variable is different for each test, so it is ok to reuse it.

```suggestion
            const erc1271Contract = randomEthereumAddress()
            await subscriber.grantPermissions(StreamPartIDUtils.getStreamID(streamPartId), {
                permissions: [StreamPermission.SUBSCRIBE],
                user: erc1271Contract
            })
            environment.getChain().erc1271AllowedAddresses.add(erc1271Contract, toEthereumAddress(subscriberWallet.address))

            const groupKey = GroupKey.generate()
            const publisher = environment.createClient({
                auth: {
                    privateKey: publisherWallet.privateKey
                }
            })
            await publisher.addEncryptionKey(groupKey, toEthereumAddress(publisherWallet.address))

            await subscriber.subscribe({
                id: StreamPartIDUtils.getStreamID(streamPartId),
                partition: StreamPartIDUtils.getStreamPartition(streamPartId),
                erc1271Contract
            }, () => {})

            await triggerGroupKeyRequest(groupKey, publisher)

            const request = await environment.getNetwork().waitForSentMessage({
                messageType: StreamMessageType.GROUP_KEY_REQUEST
            })
            await assertGroupKeyRequest(request, [groupKey.id], erc1271Contract, SignatureType.ERC_1271)
            const keyStore = getLocalGroupKeyStore(await subscriber.getAddress())
            await waitForCondition(async () => (await keyStore.get(groupKey.id, toEthereumAddress(publisherWallet.address))) !== undefined)
```

Reply:
The `subscribe` creates the stream therefore it is granted all permissions by default. In the test I want to be extra sure that the subscriber using the erc1271contract could not be receiving messages just because it was the creator of the stream. That's why I create a subscriber2 that is "truly" an outsider.